wire-runtime/src/commonMain/kotlin/com/squareup/wire/MessageSource.kt

```diff
@@ -0,0 +1,24 @@
+/*
+ * Copyright 2019 Square Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.squareup.wire
+
+expect interface MessageSource<out T> {
+  fun receiveOrNull(): T?
```

Comment:
Definitely needs docs

Reply:
Should this just be `read()`? The OrNull suffix is most useful in Java where it's otherwise unclear that null is a possibility.

Reply:
Late review, but should `T` have an upper bound of `Any`? Otherwise `null` is ambiguous if `T` is nullable. And if `T` _should_ be nullable, then this API is inherently problematic. I'm assuming that `null` here is intended to indicate the source is closed or something. Channels are (eventually) getting [`receiveOrClosed`](https://github.com/Kotlin/kotlinx.coroutines/commit/a8904e2b6fb774c251e491c05ffcc1ee2a2cfe2c#diff-bb85d2a203a08f1a61323b053fa20e87R268) to solve this same problem with their API.

Reply:
https://github.com/square/wire/pull/1088